contracts/src/v0.8/ccip/test/pools/GHO/GhoTokenPoolRemoteE2E.t.sol

```diff
@@ -0,0 +1,446 @@
+
+    bytes32[] memory hashedMessages = new bytes32[](3);
+    hashedMessages[0] = messages[0]._hash(metaDataHash);
+    messages[0].messageId = hashedMessages[0];
+    hashedMessages[1] = messages[1]._hash(metaDataHash);
+    messages[1].messageId = hashedMessages[1];
+    hashedMessages[2] = messages[2]._hash(metaDataHash);
+    messages[2].messageId = hashedMessages[2];
+
+    bytes32[] memory merkleRoots = new bytes32[](1);
+    merkleRoots[0] = MerkleHelper.getMerkleRoot(hashedMessages);
+
+    address[] memory onRamps = new address[](1);
+    onRamps[0] = ON_RAMP_ADDRESS;
+
+    bytes memory commitReport = abi.encode(
+      CommitStore.CommitReport({
+        priceUpdates: _getEmptyPriceUpdates(),
+        interval: CommitStore.Interval(messages[0].sequenceNumber, messages[2].sequenceNumber),
+        merkleRoot: merkleRoots[0]
+      })
+    );
+
+    vm.resumeGasMetering();
+    s_commitStore.report(commitReport, ++s_latestEpochAndRound);
+    vm.pauseGasMetering();
+
+    vm.mockCall(
+      s_commitStore.getStaticConfig().rmnProxy,
+      abi.encodeWithSelector(IRMN.isBlessed.selector, IRMN.TaggedRoot(address(s_commitStore), merkleRoots[0])),
+      abi.encode(true)
+    );
+
+    bytes32[] memory proofs = new bytes32[](0);
+    uint256 timestamp = s_commitStore.verify(merkleRoots, proofs, 2 ** 2 - 1);
+    assertEq(BLOCK_TIME, timestamp);
+
+    // We change the block time so when execute would e.g. use the current
+    // block time instead of the committed block time the value would be
+    // incorrect in the checks below.
+    vm.warp(BLOCK_TIME + 2000);
+
+    vm.expectEmit();
+    emit EVM2EVMOffRamp.ExecutionStateChanged(
+      messages[0].sequenceNumber,
+      messages[0].messageId,
+      Internal.MessageExecutionState.SUCCESS,
+      ""
+    );
+
+    vm.expectEmit();
+    emit EVM2EVMOffRamp.ExecutionStateChanged(
+      messages[1].sequenceNumber,
+      messages[1].messageId,
+      Internal.MessageExecutionState.SUCCESS,
+      ""
+    );
+
+    vm.expectEmit();
+    emit EVM2EVMOffRamp.ExecutionStateChanged(
+      messages[2].sequenceNumber,
+      messages[2].messageId,
+      Internal.MessageExecutionState.SUCCESS,
+      ""
+    );
+
+    Internal.ExecutionReport memory execReport = _generateReportFromMessages(messages);
+
+    uint256 preGhoTokenBalanceUser = dstGhoToken.balanceOf(USER);
+
+    vm.resumeGasMetering();
+    s_offRamp.execute(execReport, new EVM2EVMOffRamp.GasLimitOverride[](0));
+    vm.pauseGasMetering();
+
+    assertEq(preGhoTokenBalanceUser + 6000 * 1e18, dstGhoToken.balanceOf(USER), "Wrong balance on destination");
+    assertEq(dstGhoTokenPool.getCurrentBridgedAmount(), 0);
+  }
+
+  function testRevertRateLimitReached() public {
+    RateLimiter.Config memory rateLimiterConfig = _getOutboundRateLimiterConfig();
+
+    // will revert due to rate limit of tokenPool
+    sendRequestGho(1, rateLimiterConfig.capacity + 1, true, false);
+
+    // max capacity, won't revert
```

Comment:
Is there supposed to be a call to `sendRequestGho` after this? I think it should correspond to the next one. Maybe can change the spacing and put it right before the `vm.startPrank(OWNER)` line

Reply:
no there's not, it's supposed to convey now we're checking the happy case when there's sufficient capacity available to send gho by inflating facilitator level below (which is `_inflateFacilitatorLevel`)